Python/sysmodule.c

```diff
@@ -1878,7 +1878,15 @@ _PySys_GetSizeOf(PyObject *o)
         return (size_t)-1;
     }
 
-    return (size_t)size + _PyType_PreHeaderSize(Py_TYPE(o));
+    size_t presize = 0;
+    if (!Py_IS_TYPE(o, &PyType_Type) ||
+         PyType_HasFeature((PyTypeObject *)o, Py_TPFLAGS_HEAPTYPE))
+    {
+        /* Add the size of the pre-header if "o" is not a static type */
+        presize = _PyType_PreHeaderSize(Py_TYPE(o));
+    }
```

Comment:
This now more closely matches the behavior of the Python [`check_sizeof`](https://github.com/python/cpython/blob/b52fc70d1ab3be7866ab71065bae61a03a28bfae/Lib/test/support/__init__.py#L830-L833) function, although that function does not account for space for managed dict/weakref.